### DISCUSSION TITLE
Check complete page is not showing order information

### DISCUSSION THREAD

**Discussion by freshface** [duncanmcclean/simple-commerce#1180] [Help]
After doing a successful payment i get redirected back to the complete route and view: https://lively-pebble-mx3go0qkyw.ploi.dev/checkout/complete

But no info is shown? Is this session based? Or should there be a parameter in the url?

Ps, i did not visit the url. This is a screenshot after being redirected from mollie :)

<img width="1104" alt="Screenshot 2024-11-21 at 20 20 57" src="https://github.com/user-attachments/assets/eb09738c-fa1a-4967-96e8-1ef28b0ef064">

**Comment by freshface** [comment]
Update; could an https issue in the webhook.
i am using ploi's test domain name.

Just added a tenant, now waiting to resolve the dns.

**Comment by duncanmcclean** [comment]
Hmm, it could be that it's redirecting you to a different domain than your cart was originally created on? 

The cart is assigned by a cookie - and when you visit that page, it'll read the cookie, return the info, then delete the cookie.

**Comment by freshface** [comment]
Its on the same domain.
I used tenant know and have the same issue.
This is the url with products https://dewarmstecake.figure8.be/producten

**Reply by duncanmcclean** [reply]
Can you try adding this to your Simple Commerce config:

```php
// config/simple-commerce.php

'disable_form_parameter_validation' => true,
```

Then, on the checkout page, find the `<form>` that takes you to Mollie and inspect it. At the top of the form, you should see hidden inputs like this, only they'll be decrypted.

![CleanShot 2024-11-21 at 20 01 12](https://github.com/user-attachments/assets/3617c2a2-85f0-43fe-8b95-c38487011169)

Can you show what the value of the `_redirect` property is?

**Comment by freshface** [comment]
<img width="668" alt="Screenshot 2024-11-21 at 21 16 27" src="https://github.com/user-attachments/assets/e40025d7-4b3c-434b-87fb-fb67737f0da3">

**Reply by duncanmcclean** [reply]
Hmm, that looks correct. I'm not quite sure why it's not working as expected. 🤔

I'm currently focused on the [next version of Simple Commerce](https://duncanmcclean.com/future-of-simple-commerce) which includes some changes to how carts/checkout is handled. It'll probably have more robust handling for the post-checkout page.

**Comment by freshface** [comment]
After being redirected the cookie is not available anymore.
Is there a way to use the newer version? Or still to much in beta?

**Reply by duncanmcclean** [reply]
> After being redirected the cookie is not available anymore.

Just had a look at the code....

It looks like the cookie *should* be deleted, but there should be a `simple-commerce.checkout.success` key in the user's session which indicates if the checkout was successful (you won't be able to see it via browser dev tools).

I've just given the Mollie checkout process a try here and it worked successfully:

![CleanShot 2024-11-23 at 00 08 36](https://github.com/user-attachments/assets/221f8ebc-dae0-4808-8bcd-59c523c224b8)

Are there any errors in the Mollie payment's history timeline re the webhook? What's `APP_URL` set to in your `.env`?

> Is there a way to use the newer version? Or still to much in beta?

Not yet - it's still mid-way through development.

**Comment by freshface** [comment]
No errors in mollie dashboard.
Will try in a clean statamic.

**Comment by freshface** [comment]
I did a clean install.
Same issue.

**Reply by duncanmcclean** [reply]
Can you add a `{{ dump }}` to the cart complete page and let me know what you see?

**Reply by duncanmcclean** [reply]
Also, are you storing orders as entries or in the database?

**Comment by freshface** [comment]
Yes, i am storing in a DB (mysql)

Dump

```
array:39 [▼ // vendor/statamic/cms/src/Tags/Dump.php:16
  "browser_appearance" => Statamic\Globals\Variables {#1138 ▶}
  "config" => array:32 [▶]
  "configuration" => Statamic\Globals\Variables {#1132 ▶}
  "cp_url" => "https://dewarmstecake.figure8.be/cp"
  "csrf_field" => Illuminate\Support\HtmlString {#707 ▶}
  "csrf_token" => "ow9C846WBqcVXGnR8kzHtoMu1k33r0iTJjfwnwXT"
  "current_date" => Illuminate\Support\Carbon @1732622012 {[#1079 ▶](https://dewarmstecake.figure8.be/checkout/complete#sf-dump-1872487206-ref21079)}
  "current_full_url" => "https://dewarmstecake.figure8.be/checkout/complete"
  "current_layout" => "layout"
  "current_template" => "checkout.complete"
  "current_uri" => "/checkout/complete"
  "current_url" => "https://dewarmstecake.figure8.be/checkout/complete"
  "current_user" => Statamic\Auth\File\User {#1081 ▶}
  "environment" => "local"
  "errors" => Illuminate\Support\ViewErrorBag {#1012 ▶}
  "get" => []
  "get_post" => []
  "homepage" => "/"
  "is_homepage" => false
  "last_segment" => "complete"
  "logged_in" => true
  "logged_out" => false
  "now" => Illuminate\Support\Carbon @1732622012 {[#1079 ▶](https://dewarmstecake.figure8.be/checkout/complete#sf-dump-1872487206-ref21079)}
  "old" => []
  "post" => []
  "redirects" => Statamic\Globals\Variables {#1158 ▶}
  "response_code" => 200
  "segment_0" => ""
  "segment_1" => "checkout"
  "segment_2" => "complete"
  "seo" => Statamic\Globals\Variables {#1144 ▶}
  "settings" => Statamic\Globals\Variables {#1128 ▶}
  "site" => Statamic\Sites\Site {[#612 ▶](https://dewarmstecake.figure8.be/checkout/complete#sf-dump-1872487206-ref2612)}
  "sites" => Illuminate\Support\Collection {#1073 ▶}
  "social_media" => Statamic\Globals\Variables {#1124 ▶}
  "title" => "Checkout"
  "today" => Illuminate\Support\Carbon @1732622012 {[#1079 ▶](https://dewarmstecake.figure8.be/checkout/complete#sf-dump-1872487206-ref21079)}
  "view" => []
  "xml_header" => "<?xml version="1.0" encoding="utf-8" ?>"
]```

**Reply by freshface** [reply]
You can try it yourself on https://dewarmstecake.figure8.be

**Reply by duncanmcclean** [reply]
Thanks! I'll take a look when I can.

**Reply by duncanmcclean** [reply]
I've just tested this again, both on my existing test app and on a fresh install w/ database orders, and I'm still unable to reproduce the issue. The order information is shown correctly:

![CleanShot 2024-12-07 at 19 13 36](https://github.com/user-attachments/assets/26ecb2d0-160b-4611-888e-5eafbb282ff1)

Can you try `{{ session:dump }}`? It should output a `simple-commerce` key like this:

![CleanShot 2024-12-07 at 19 15 20](https://github.com/user-attachments/assets/b17bac7a-c297-4b08-b2dc-de59fb28a2f5)
